Версия 1

Akmula opened the pull request:
1. Создать пакет model. а. Добавить класс Film со следующими свойствами:
		* целочисленный идентификатор — id;
		* название — name;
		* описание — description;
		* дата релиза — releaseDate;
		* продолжительность фильма — duration. б. Добавить класс User со следующими свойствами:
		* целочисленный идентификатор — id;
		* электронная почта — email;
		* логин пользователя — login;
		* имя для отображения — name;
		* дата рождения — birthday.

2. Создать класс-контроллер FilmController а. Добавить эндпоинты:
		* добавление фильма;
		* обновление фильма;
		* получение всех фильмов.

3. Создать класс-контроллер UserController а. Добавить эндпоинты:
		* создание пользователя;
		* обновление пользователя;
		* получение списка всех пользователей.

4. Добавить валидацию данных: а. для Film:
		* название не может быть пустым;
		* максимальная длина описания — 200 символов;
		* дата релиза — не раньше 28 декабря 1895 года;
		* продолжительность фильма должна быть положительным числом. б. для User:
		* электронная почта не может быть пустой и должна содержать символ @;
		* логин не может быть пустым и содержать пробелы;
		* имя для отображения может быть пустым — в таком случае будет использован логин;
		* дата рождения не может быть в будущем.

5. Добавить unit-тесты для валидации моделей.

6. Добавить в описание сборки проекта зависимость "spring-boot-starter-validation" и применить аннотации к полям классов-моделей.

Akmula committed:
Версия 1
1. Создать пакет model.
	а. Добавить класс Film со следующими свойствами:
		* целочисленный идентификатор — id;
		* название — name;
		* описание — description;
		* дата релиза — releaseDate;
		* продолжительность фильма — duration.
	б. Добавить класс User со следующими свойствами:
		* целочисленный идентификатор — id;
		* электронная почта — email;
		* логин пользователя — login;
		* имя для отображения — name;
		* дата рождения — birthday.

2. Создать класс-контроллер FilmController
	а. Добавить эндпоинты:
		* добавление фильма;
		* обновление фильма;
		* получение всех фильмов.

3. Создать класс-контроллер UserController
	а. Добавить эндпоинты:
		* создание пользователя;
		* обновление пользователя;
		* получение списка всех пользователей.

4. Добавить валидацию данных:
	а. для Film:
		* название не может быть пустым;
		* максимальная длина описания — 200 символов;
		* дата релиза — не раньше 28 декабря 1895 года;
		* продолжительность фильма должна быть положительным числом.
	б. для User:
		* электронная почта не может быть пустой и должна содержать символ @;
		* логин не может быть пустым и содержать пробелы;
		* имя для отображения может быть пустым — в таком случае будет использован логин;
		* дата рождения не может быть в будущем.

5. Добавить unit-тесты для валидации моделей.

6. Добавить в описание сборки проекта зависимость "spring-boot-starter-validation" и применить аннотации к полям классов-моделей.

YazykovMikhail commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -1,7 +1,65 @@
+        if (films.containsKey(newFilm.getId())) {
+            Film oldFilm = films.get(newFilm.getId());
+            validationReleaseDate(newFilm);
+            oldFilm.setName(newFilm.getName());
```
Просто полностью обновляй обьект:
		films.put(newFilm.getId(), newFilm);

YazykovMikhail replied:
Касается и User

Akmula replied:
Исправлено

YazykovMikhail commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -1,7 +1,65 @@
+import java.util.HashMap;
+import java.util.Map;
+
+@Slf4j
```
Стандартный шаблон для логов
```
createMethod(Request request) {
log.info("пришел Post запрос /films с телом: {}", request)
Response response = //логика метода, если есть бизнес процессы для логирвоания логируем их
log.info("Отправлен ответ Post /films с телом: {}", response)
}
```
что бы было видно что пришло, что ушло и если были какие то изменения в процессы выполнянения тоже были видны, потому что ты пойдешь в логи когда у тебя что то сломалось, просто по сообщения "Фильм добалвен" ничего не понятно.

Akmula replied:
Исправил. Но не совсем понял, как это делать)

YazykovMikhail replied:
А что именно? Должен быть залогирвоан входящий запрос и ответ, что бы можно было потом по логам понять что было или найти проблему

Akmula replied:
Попытался исправить

YazykovMikhail commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -1,7 +1,65 @@
+        }
+    }
+
+    private int getNextId() {
```
Представь что у тебя 8767890 фильмов и при сощдании нового придется по всем пробежаться и высчитать масимальное id - сложно
Добавь поле которое булет счетчиком и увеличивай его, так же как было в TaskManager

Akmula replied:
Исправлено

YazykovMikhail commented on src/main/java/ru/yandex/practicum/filmorate/model/Film.java:
```diff
@@ -1,12 +1,28 @@
+
+    private Integer id;
+
+    @NotEmpty(message = "Название не может быть пустым!")
```
Лучше @NotBlank

Akmula replied:
Исправлено

Akmula committed:
Версия 1.1
1. Изменить метод обновления User и Film.
2. Добавить логирование.
3. Изменить генерацию ID
4. Изменить аннотацию в Film у поля name на @NotBlank

YazykovMikhail commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -1,7 +1,68 @@
+    @PostMapping
+    public Film createFilm(@Valid @RequestBody Film film) {
+        log.info("Запрос POST /films - добавить фильм: {}", film);
+        log.debug("Валидация даты релиза: {}", film.getReleaseDate());
```
это лишнее, в верхнем логе уже есть эта информация

Akmula replied:
Удалил

YazykovMikhail commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -1,7 +1,68 @@
+@RequestMapping("/films")
 public class FilmController {
-}
+    protected Integer filmId = 0;
```
private, protected не за чем

Akmula replied:
Изменил

YazykovMikhail commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -1,7 +1,68 @@
+    public Collection<Film> getFilms() {
+        log.info("GET /films - получение всех фильмов.");
+        return films.values();
+    }
```
Тут ответ не залогирован

Akmula replied:
Исправил
